tests/v3_validation/cattlevalidationtest/core/test_hosttemplate.py

```diff
@@ -0,0 +1,550 @@
+from common_fixtures import *  # NOQA
+from test_machine import get_droplets
+import gdapi
+import logging
+import machine
+import time
+
+DEFAULT_TIMEOUT = 900
+
+# Digital Ocean configurations
+access_token = os.environ.get('DIGITALOCEAN_KEY')
+image_name = "ubuntu-16-04-x64"
+region = "sfo1"
+size = "1gb"
+
+# Amazon EC2 configurations
+access_key = os.environ.get('AMAZON_ACCESSKEY')
+secret_key = os.environ.get('AMAZON_SECRETKEY')
+
+# Digital Ocean default configurations
+default_size = "512mb"
+default_image_name = "ubuntu-16-04-x64"
+default_region = "nyc3"
+
+# Digital Ocean Error Messages
+error_msg_auth_failure = "Invalid access token"
+
+error_msg_invalid_region = "digitalocean requires a valid region"
+
+if_machine_digocean = pytest.mark.skipif(
+    not os.environ.get('DIGITALOCEAN_KEY'),
+    reason='DIGITALOCEAN_KEY is not set')
+if_machine_amazon = pytest.mark.skipif(
+    not os.environ.get('AMAZON_ACCESSKEY') or
+    not os.environ.get('AMAZON_SECRETKEY'),
+    reason='Amazon EC2 Accesskey/Secretkey is not set')
+
+# Get logger
+logger = logging.getLogger(__name__)
+
+
+def create_ht(ht_create_args, client):
+    ht_list = client.list_hostTemplate()
+    len_ht_list = len(ht_list.data)
+
+    ht0 = client.wait_success(client.create_hostTemplate(**ht_create_args))
+
+    assert ht_create_args["name"] == ht0.name
+    assert ht0.id is not None and ht0.id != ""
+    ht_list = client.list_hostTemplate()
+    assert len(ht_list.data) == len_ht_list + 1
+    assert len(filter(lambda ht: ht.id == ht0.id, ht_list.data)) == 1
+    return ht0
+
+
+def cleanupht(ht_to_remove, client):
```

Comment:
really, should be cleanup_host_template. You can use the `ht` abbreviation in variable names, but should avoid it in function names